tests/test_dagviz.py

```diff
@@ -0,0 +1,100 @@
+'''
+Tests for dagviz and FlowImage class.
+'''
+
+import pytest
+import io
+import warnings
+import unittest.mock as mock
+import xml
+from xml.etree import ElementTree as ET
+import pydot
+import networkx as nx
+import PIL
+import bionic.dagviz as dagviz
+
+
+@pytest.fixture
+def pydot_graph():
+    '''Create empty pydot graph for testing'''
+    return pydot.Dot(graph_type='graph')
+
+
+@pytest.fixture
+def flow_image(pydot_graph):
+    '''Create FlowImage fixture for testing'''
+    return dagviz.FlowImage(pydot_graph)
+
+
+def test_flowimage_typing(flow_image):
+    '''Check types for attributes of flowImage'''
+    display_str = flow_image._repr_svg_()
+    try:
+        ET.fromstring(flow_image.xml_text)
+    except xml.etree.ElementTree.ParseError:
+        pytest.fail("FlowImage.xml_text not well formed XML")
```

Comment:
It might be helpful to include both the parsing exception and the XML text in the failure message here. Maybe something like this?

```
except xml.etree.ElementTree.ParseError as e:
    raise AssertionError(f"FlowImage.xml_text not well formed XML {flow_image.xml_text!r}") from e
```

Reply:
Added string interpolation to the `pytest.fail`